### ISSUE TITLE
SizeOf() changes with newer dotnet

### ISSUE THREAD

**Issue by Zoxive** [dbolin/Apex.Runtime#49]
We have some new interesting projects like supporting mobile that doesnt allow Reflection.Emit so I have been experimenting with seeing if i can come up with an alternative code path when `RuntimeFeature.IsDynamicCodeSupported` is false. I found a few promising ideas. 

Not sure your plans with this library, but i noticed while testing against netcoreapp3.1 or net5 that `string`, `ValueTask` and `Task` sizes are different. I guess thats kind of expected that the sizes will change since the dotnet itself doesnt expose this information.

Just thought i would see if you knew about these changes incase you were unaware.

**Comment by dbolin** [comment]
Are you thinking of AOT code gen?  I've considered it, but .NET hasn't had a lot of good options there.  Generics can be pretty tricky to handle in a performant way, and for serialization readonly fields will be another issue as well.

I didn't know about the changes, but it makes sense (they're all smaller).  I've published a new version that specifically targets .net 5 with the new computations.

**Comment by Zoxive** [comment]
Not exactly AOT.. basically just need to support xamarin. 

Which has limitations. Specificly Reflection.Emit. https://docs.microsoft.com/en-us/xamarin/ios/internals/limitations#systemreflectionemit

I made a fork with what i think im going to run with. https://github.com/Zoxive/Apex.Runtime
I changed the namespace to prevent collisions, but at this time i dont plan on publishing this to nuget. (Just using an private nuget feed for now)

The main difference is using the objects MethodTable to get the size, which is more of a hack than Apex.Runtime implementation.. but it was the only option I saw.
